### ISSUE TITLE
Trezor1 not signing mixed taproot+wpkh inputs?

### ISSUE THREAD

**Issue by jb55** [bitcoin-core/HWI#585]
Using bitcoin-core qt's coin selection I created 3 PSBTs: one with 3 inputs, one with 2 inputs, and one with 1 input.

In all three cases, only 1 input is signed, but individually they are signed just fine. attemping to sign the PSBT multiple times does nothing.

I don't remember this ever being an issue? perhaps something changed?

`device` `trezor1` `1.10.5`
`hwi version` `3fe369d0379212fae1c72729a179d133b0adc872`

input paths are: 

`tr([4......b/86'/0'/0'/0/1])`
`wpkh([4......b/84'/0'/4'/0/34])`
`wpkh([4......b/84'/0'/4'/0/27])`

it outputs a PSBT with `partial_signatures` for the `m/84'/0'/4'/0/34` input and nothing else?

any idea what's going on here?

**Comment by achow101** [comment]
It is only signing that one particular wpkh input?

**Comment by jb55** [comment]
correct

**Comment by achow101** [comment]
Does the PSBT with 2 inputs have the two wpkhs or the one wpkh and tr?

**Comment by jb55** [comment]
Tried with only the 2 wpkh inputs and it still only signs the `m/84'/0'/4'/0/34` one.

Tried without  `m/84'/0'/4'/0/34`, with tr and `wpkh([4......b/84'/0'/4'/0/27])`, that one failed to sign any inputs

Tried `wpkh([4......b/84'/0'/4'/0/27])` by itself and it fails entirely (wtf? I've tried multiple times to confirm this as well)

Tried `wpkh(m/84'/0'/4'/0/34)` + `tr([4......b/86'/0'/0'/0/1])` and it only signs `m/84'/0'/4'/0/34`

Tried `tr` by itself and fails.

In conclusion it looks like it can't sign either `wpkh([4......b/84'/0'/4'/0/27])` or `tr([4......b/86'/0'/0'/0/1])`

uhh??

**Comment by jb55** [comment]
> Also, unless you have a build of Bitcoin Core with https://github.com/bitcoin/bitcoin/pull/22558, the tr input will not be signed.

ohhhh good to know

**Comment by achow101** [comment]
Does the `wpkh([4......b/84'/0'/4'/0/27])` input have BIP 32 derivation paths in the PSBT and do they look right?

**Comment by jb55** [comment]
oh nm that one was actually a taproot output as well, I was confused by the GUI!

I don't care about doxing myself here since there are just dust inputs that I was playing with when testing HWI+taproot stuff

wpkh -> taproot
https://blockstream.info/address/bc1q38cqrsmgfdp66cct2yq8eump3fn6sx2njzdpf3

tree-mode coinselection gui was showing me wpkh which was throwing me off, this is what I was looking at:
![Mar03-114350](https://user-images.githubusercontent.com/45598/156640976-ca1c3635-79bf-4bec-afb0-7a55c0d2b620.png)

the real output:
![Mar03-114039](https://user-images.githubusercontent.com/45598/156640387-2db40028-ae37-45d5-a1ca-5e35238fe52a.png)

so that broken `wpkh` was actually just a `tr([446284ab/86'/0'/0'/0/0]` all along. mystery solved! I'll try your taproot input PR for core and close this for now.

**Comment by jb55** [comment]
@achow101 so I updated to that PR, curiously when I sign those two taproot inputs individually and together it's fine, but when I mix in the wpkh input I get this error:

```
{"error": "Previous transaction 7792d208f7f4bbc789ce17a3d614d87b29e04ede7c953afb434dd41a03cc908e not available", "code": -7}
```

and the trezor signing bar freezes. perhaps this is a trezor1 bug? @prusnak 

* tr + tr work
* wpkh works
* tr + tr + wpkh fails

**Comment by prusnak** [comment]
@matejcik can you please have a look at this?

**Comment by prusnak** [comment]
@jb55 For non-Taproot inputs (including segwit v0) you need to provide prevtx in PSBT so Trezor can confirm the input amount. For Taproot this is not required because Taproot inputs commit to all input amounts.

**Comment by achow101** [comment]
Have you checked out the latest version of the Taproot fields PR? There was an older version (several months ago) which dropped the non_witness_utxo for any segwit input which lead to an error like the one you see. The latest version of the PR shouldn't do that anymore.

**Comment by jb55** [comment]
@achow101 derp you're right I might have pulled an older branch. will try again

**Comment by jb55** [comment]
sorry for the noise, will be back if I have any issues lol
